### PR TITLE
Use classes instead of objects and arrays

### PR DESCRIPTION
**Pull request by martinkarugaba**
## Because my coding partner has opted out of the program. I have been given permission to continue in this Block solo. If you have any questions please reach out to Marco at Student Success: [success@microverse.org](mailto:success@microverse.org)

### 
continue_solo

# What I have worked on in this Pull Request

- [ ] Used JavaScript classes instead of objects and arrays.
- [ ] Maintained all of the original functionality of the application.
- [ ] Added CSS styles to the application to make it match the wireframe.

### CODE REVIEW

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @martinkarugaba, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that doesn't have an updated readme and also doesn't have commis from @Amrit131. Kindly make sure both contribute in order to move forward. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello  :wave: ,
I am Nemwel your reviewer for this PR :man_beard: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## :heavy_multiplication_x:  Invalid Code Review Request :heavy_multiplication_x: 

As requested by the previous reviewer, kindly ensure that you use the new and updated Microverse README template in your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,62 +1,127 @@
...
-Give a ⭐️ if you like this project!
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-## Acknowledgments
+## 📝 License <a name="license"></a>
 
-- Hat tip to anyone whose code was used
-- Inspiration
-- etc
+This project is [MIT](./LICENSE) licensed.
```
Comment (HAMMAS-SALEEM) [inline README.md:128]: - [ ] This link is broken in your readme file. Kindly fix it to make your readme file descriptive :+1:

File: books.js [hunk 2]
```diff
@@ -1,62 +1,66 @@
...
-const addBtn = document.getElementsByClassName('add-btn')[0];
-const bookSection = document.getElementsByClassName('books')[0];
-let books = [];
+/* eslint-disable max-classes-per-file */
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
```
Comment (HAMMAS-SALEEM) [inline books.js]: - [ ] I saw that you are not using ESLint in your linters setup. ESLint helps you to identify errors in your JavaScript files. These errors can break your whole application :fearful: Kindly use ESLint in your linters setup :+1:

Comment (HAMMAS-SALEEM) [inline books.js]: - [ ] You said that you are allowed to submit this activity without your partner commits. Can attach a screenshot of the email(In which you got permission from StudentSuccessTeam) in your pull request :blush:

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:
invalid

## Hey @martinkarugaba :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Github flow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Invalid Code Review Request ❌

Hi @martinkarugaba, This is @sja-thedude your code reviewer.

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

You have submitted a project, that doesn't have the correct linters setup. Use JS linters from [here](https://github.com/microverseinc/linters-config/tree/master/javascript) and delete your current linters.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: js/books.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    booksArr = booksArr.filter((item) => item.id !== +id);
+    Storage.addToStorage(booksArr);
+  }
+}
+
+window.addEventListener('DOMContentLoaded', () => {
+  UI.displayData();
+  UI.removeBook();
+});
+
```
Comment (BenMukebo) [inline js/books.js:50]: - [ ] Great job adding the `methods` to add and remove books. However the `add` and `remove` are not working 😮‍💨 Please make sure the add and remove bookwork as expected. 💯 👍🏻 

[Screencast from 01-12-2023 06:47:43 PM.webm](https://user-images.githubusercontent.com/79058364/212130721-e8cee5ba-9683-45ef-b8f1-18bf6a5ba52d.webm)

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @martinkarugaba, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- Correctly setup JS linters. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Your app styles match [this wireframe](https://github.com/microverseinc/curriculum-javascript/blob/bb245ca8a9d3bb04ebefbbd45e800764e1fc5f74/books/images/awesome_books_core_elements.png) ✔️


 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by martinkarugaba** [comment]
Hi @BenMukebo, this error is because in books.js file, while importing the two modules, `book` and `localStorage` eslint does not want me to add file extension `.js` to the file. When I add id it returns an eslint error, how would you advise me to go about it?

**Review by codecaiine — APPROVED** [review]
Hi @martinkarugaba  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great project implementation :heavy_check_mark: 
- Good use Github flow :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**